triton-vm/src/table/constraint_circuit.rs

```diff
@@ -383,7 +462,10 @@ impl<T: TableChallenges> ConstraintCircuit<T> {
                     }
                 }
             }
-            CircuitExpression::Input(index) => max_degrees[*index],
+            CircuitExpression::Input(input) => {
+                let index: usize = (*input).into();
+                max_degrees[index]
```

Comment:
You asked when casting to/from `usize` was relevant for this data type. One example of converting *to* `usize where I don't think it can be avoided is when the symbolic degree bounds are found.